src/client.rs

```diff
@@ -57,9 +57,10 @@
 //! returns `Ready`. At this point, the underlying socket has been closed and no
 //! further work needs to be done.
 //!
-//! The easiest option is to submit the [`Connection`] instance to an
-//! [executor]. The executor will then manage polling the connection until the
-//! connection is complete.
+//! The easiest way to ensure that the [`Connection`] instance gets polled is to
+//! submit the [`Connection`] instance to an [executor]. The executor will then
+//! manage polling the connection until the connection is complete.
+//! Alternatively, the caller can call `poll` manually.
```

Comment:
This might be too fine-grained, so TIOLI, but should we give any reason _why_ you might `poll` manually? I  feel like, when something is typically only done in a few specific cases, enumerating what those cases might be is a good way to discourage users from doing that thing when it's not necessary. If that makes sense.